### PR TITLE
Anko - Handle onItemClick function for  list invite screen

### PR DESCRIPTION
**Pull request by at-ngoctran**
#### Related issues (Required)
- Task #153 

#### Description (Required)
- [x] Handle onItemClick function for list invite

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt [hunk 1]
```diff
@@ -34,6 +36,12 @@ class InviteViewModel(val context: Context) {
                 }
     }
 
+    internal fun inviteUserJoinToGroup(userId: String, groupId: String, groupName: String,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/bc39d288676bd65b4e1171a96fcfdb6a6880f673/app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt#L39"></span>Unexpected indentation</td>
    </tr>
  </tbody>
</table>

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

File: app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt [hunk 2]
```diff
@@ -34,6 +36,12 @@ class InviteViewModel(val context: Context) {
...
+                                       ownerId: String, userInvited: User) {
+        val inviteRef = FirebaseDatabase.getInstance().getReference("user/${userInvited.id}/invites/$groupId")
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt]: Should not put data source in viewModel

Comment (at-tienhoang) [reply app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt]: Do we need to handle something after invite?

File: app/src/main/java/vn/asiantech/way/ui/group/invite/InviteFragment.kt [hunk 3]
```diff
@@ -74,9 +72,8 @@ class InviteFragment : BaseFragment() {
     /**
      * On item invite click of  RecyclerView list.
      */
-    internal fun eventItemInviteClicked(user: User) {
-        val inviteRef = FirebaseDatabase.getInstance().getReference("user/${user.id}/invites/$groupId")
-        inviteRef.setValue(Invite(userId, groupId, groupName, userId == ownerId))
+    internal fun eventItemInviteClicked(userInvited: User) {
+        viewModel.inviteUserJoinToGroup(userId, groupId, groupName, ownerId, userInvited)
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/invite/InviteFragment.kt]: Please keep `userId, groupId, groupName, ownerId` in viewmodel

File: app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt [hunk 4]
```diff
@@ -1,10 +1,12 @@
 package vn.asiantech.way.ui.group.invite
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt]: Please add second constructor for unit test

Comment (at-tienhoang) [reply app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt]: Please remove `.observeOnUiThread()` in viewmodel

File: app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt (hunk 2 above)
Comment (at-ngoctran) [reply app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt]: Do we need to handle something after invite?
No, I follow to logic of old file

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 5]
```diff
@@ -44,8 +44,8 @@ class GroupRepository : GroupDataSource {
         return remoteDataSource.removeGroup(groupId)
     }
 
-    override fun postInvite(userId: String, invite: Invite): Single<Boolean> {
-        return remoteDataSource.postInvite(userId, invite)
+    override fun inviteUserJoinGroup(userId: String, invite: Invite): Single<Boolean> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/e0a555ff8a9c7a3b95f6a6641e5f8281958a20d4/app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt#L47"></span></td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt [hunk 6]
```diff
@@ -3,29 +3,30 @@ package vn.asiantech.way.ui.group.invite
...
-import vn.asiantech.way.extension.observeOnUiThread
 import vn.asiantech.way.utils.AppConstants
 import java.util.concurrent.TimeUnit
 
 /**
  * Invite View Model
  * @author NgocTTN
  */
-class InviteViewModel(val context: Context) {
+class InviteViewModel(val context: Context, private val groupRepository: GroupRepository) {
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt]: Remove context

File: app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt [hunk 7]
```diff
@@ -34,10 +35,14 @@ class InviteViewModel(val context: Context) {
...
+    internal fun inviteUserJoinToGroup(userId: String, invite: Invite) {
+        groupRepository.inviteUserJoinGroup(userId, invite)
+    }
+
     private fun getListUser(name: String): Observable<List<User>> {
         return groupRepository
                 .searchUser(name)
                 .doOnSubscribe { resetDataStatus.onNext(true) }
-                .observeOnUiThread()
+                .subscribeOn(Schedulers.io())
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/invite/InviteViewModel.kt]: move this line to searchInviteObservable

**Comment by at-dev-support-01** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please rebase to get rid of the merge commits in this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**